waiter/integration/waiter/proto_test.clj

```diff
@@ -16,69 +16,119 @@
+                (doseq [{:keys [correlation-id expected-response-size response]}
+                        (parallelize-requests
+                          10 ;; num threads
+                          5 ;; num iterations
+                          (fn []
+                            (let [correlation-id (rand-name)
+                                  kitchen-response-size (+ 200000 (* 10000 (rand-int 20)))
+                                  request-headers (-> request-headers
+                                                      (dissoc "x-cid")
+                                                      (assoc :x-cid correlation-id
+                                                             :x-kitchen-chunk-delay (rand-int 10)
+                                                             :x-kitchen-chunk-size (+ 2000 (* 100 (rand-int 20)))
+                                                             :x-kitchen-response-size kitchen-response-size
+                                                             :x-waiter-debug true))]
+                              {:correlation-id correlation-id
+                               :expected-response-size kitchen-response-size
+                               :response (make-shell-request waiter-url request-headers :path "/chunked" :protocol protocol)}))
+                          :verbose true)]
+                  (assert-streaming-response waiter-url correlation-id protocol expected-response-size response))))))))))
 
-(deftest ^:parallel ^:integration-fast test-h2c-backend-proto-service
+(deftest ^:parallel ^:integration-slow test-http-backend-proto-service
   (testing-using-waiter-url
-    (run-backend-proto-service-test waiter-url "h2c" 1 "http" "HTTP/2.0")))
+    (run-backend-proto-service-test waiter-url "http" "http" "HTTP/1.1")
+    (is "test completed marker")))
 
-(deftest ^:parallel ^:integration-fast test-h2-backend-proto-service
+(deftest ^:parallel ^:integration-slow test-https-backend-proto-service
   (testing-using-waiter-url
-    (run-backend-proto-service-test waiter-url "h2" 1 "https" "HTTP/2.0")))
+    (run-backend-proto-service-test waiter-url "https" "https" "HTTP/1.1")
+    (is "test completed marker")))
 
-(deftest ^:parallel ^:integration-fast test-h2-backend-proto-service-health-check-on-port0
+(deftest ^:parallel ^:integration-slow test-h2c-backend-proto-service
   (testing-using-waiter-url
-    (run-backend-proto-service-test waiter-url "h2" 0 "https" "HTTP/2.0")))
+    (run-backend-proto-service-test waiter-url "h2c" "http" "HTTP/2.0")
+    (is "test completed marker")))
 
-(deftest ^:parallel ^:integration-fast test-health-check-proto
+(deftest ^:parallel ^:integration-slow test-h2-backend-proto-service
   (testing-using-waiter-url
-    ;; PORT2 is running kitchen without SSL enabled
-    (run-backend-proto-service-test waiter-url "h2" "https" 1 "https" "HTTP/2.0")))
+    (run-backend-proto-service-test waiter-url "h2" "https" "HTTP/2.0")
```

Comment:
I think all 4 of these tests need to be marked `^:resource-heavy` to avoid failures on travis-k8s due to pod scaling. Unfortunately, that will also probably slow the build down significantly, since they'll all run in serial.

Reply:
Will do

Reply:
Done